vm/actor/src/builtin/market/state.rs

```diff
@@ -534,6 +528,47 @@ impl State {
     }
 }
 
+fn mutate_balance_table<BS, F>(store: &BS, c: &mut Cid, f: F) -> Result<(), String>
+where
+    F: FnOnce(&mut BalanceTable<BS>) -> Result<(), String>,
+    BS: BlockStore,
+{
+    let mut t = BalanceTable::from_root(store, &c)?;
+
+    f(&mut t)?;
+
+    *c = t.root()?;
+    Ok(())
+}
+// Unused but exists in spec
+fn _deal_proposal_is_internally_valid<BS, RT>(
+    rt: RT,
+    proposal: ClientDealProposal,
+) -> Result<(), String>
+where
+    BS: BlockStore,
+    RT: Runtime<BS>,
+{
+    if proposal.proposal.end_epoch <= proposal.proposal.start_epoch {
+        return Err(format!(
+            "proposal end epoch before start epoch: {}, {}",
+            proposal.proposal.end_epoch, proposal.proposal.start_epoch
+        ));
+    }
+
+    // Note: we do not verify the provider signature here, since this is implicit in the
+    // authenticity of the on-chain message publishing the deal.
+    let buf = proposal
+        .proposal
+        .marshal_cbor()
+        .map_err(|e| e.to_string())?;
+    rt.syscalls()
+        .verify_signature(&proposal.client_signature, &proposal.proposal.client, &buf)
+        .map_err(|e| e.to_string())?;
+
+    Ok(())
+}
```

Comment:
just remove, no need for this (can always just look back to this to apply if logic is the same)